fe1-web/src/features/evoting/network/__tests__/ElectionHandler.test.ts

```diff
@@ -69,14 +91,118 @@ beforeEach(() => {
 });
 
 afterEach(() => {
-  warn.mockClear();
+  jest.clearAllMocks();
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
 });
 
 describe('ElectionHandler', () => {
+  describe('election#setup', () => {
+    it('should return false if the object is not "election"', () => {
+      const addEvent = jest.fn();
+
+      expect(
+        handleElectionSetupMessage(addEvent)({
+          ...mockMessageData,
+          messageData: {
+            object: ObjectType.CHIRP,
+            action: ActionType.SETUP,
+          },
+        }),
+      ).toBeFalse();
+
+      expect(warn).toHaveBeenCalledTimes(1);
+      // check if the printed warning message contains substring
+      expect(warn).toHaveBeenCalledWith(
+        expect.stringMatching(/unsupported message/i),
+        expect.anything(),
+      );
+    });
+    it('should return false if the action is not "setup"', () => {
+      const addEvent = jest.fn();
+
+      expect(
+        handleElectionSetupMessage(addEvent)({
+          ...mockMessageData,
+          messageData: {
+            object: ObjectType.ELECTION,
+            action: ActionType.ADD,
+          },
+        }),
+      ).toBeFalse();
+
+      expect(warn).toHaveBeenCalledTimes(1);
+      // check if the printed warning message contains substring
+      expect(warn).toHaveBeenCalledWith(
+        expect.stringMatching(/unsupported message/i),
+        expect.anything(),
+      );
+    });
+
+    it('should create the election', () => {
+      let storedElection: ElectionState | undefined;
+
+      const addEvent = jest.fn().mockImplementation((laoId, eventState) => {
+        storedElection = eventState;
+
+        // Return a redux action, should be an action creator
+        return mockReduxAction;
+      });
+
+      expect(storedElection).toEqual(undefined);
+
+      expect(
+        handleElectionSetupMessage(addEvent)({
+          ...mockMessageData,
+          messageData: new SetupElection({
+            lao: mockLaoIdHash,
+            id: mockElectionNotStarted.id,
+            name: mockElectionNotStarted.name,
+            version: mockElectionNotStarted.version,
+            created_at: mockElectionNotStarted.createdAt,
+            start_time: mockElectionNotStarted.start,
+            end_time: mockElectionNotStarted.end,
+            questions: mockElectionNotStarted.questions,
+          }),
+        }),
+      ).toBeTrue();
+
+      // no warning should have been printed
+      expect(warn).toHaveBeenCalledTimes(0);
```

Comment:
Is it really relevant ? If at some point someone wants to write a warning in this function for whatever reason, I don't think it should make tests fail.